Changelog for the 1.3.0 release (#139)
Yoast/yoastcs: CHANGELOG.md

> This projects adheres to [Semantic Versioning](https://semver.org/) and [Keep a CHANGELOG](https://keepachangelog.com/).

Reviewer: Non-PR related, perhaps still a good place to fix this.

```suggestion
This project adheres to [Semantic Versioning](https://semver.org/) and [Keep a CHANGELOG](https://keepachangelog.com/).
```

Same for line 1 above.
```
# Change Log
```
to
```
# Changelog
```

### [1.3.0] - 2019-07-31

#### Added
* PHPCS: New `Yoast.Commenting.CoversTag` sniff.
    This sniff verifies that:
    - the contents of a `@covers` annotation is valid based on what's supported by PHPUnit;
    - there are no duplicate `@covers` or `@coversNothing` tags in a docblock;
    - a docblock doesn't contain both a `@covers` tag as well as a `@coversNothing` tag;
    Includes a fixer for common errors.
* PHPCS: New `Yoast.Commenting.TestsHaveCoversTag` sniff.
    This sniff verifies that all unit test functions have at least one `@covers` tag - or a `@coversNothing` tag - in the function docblock or in the class docblock.
* PHPCS: New `Yoast.Yoast.AlternativeFunctions` sniff.

>     This sniff allows for discouraging/forbidding the use of PHP/WP native functions in favour of using Yoast native functions.

Reviewer: We should use American spelling.

```suggestion
    This sniff allows for discouraging/forbidding the use of PHP/WP native functions in favor of using Yoast native functions.
```

    In this initial version, the sniff checks for the use of the `json_encode()` and `wp_json_encode()` functions and suggests using `WPSEO_Utils::format_json_encode()` instead.
    Note: this sniff contains an auto-fixer. If for any of the repos, the auto-fixer should not be used, the auto-fixer can be disabled from within the repo specific ruleset using `<rule ref="..." phpcs-only="true"/>`.
* PHPCS: The `Squiz.WhiteSpace.MemberVarSpacing` sniff.

>     This sniff verifies and auto-fixes the amount of blank lines between property declarations within OO-structures.

Reviewer: Blank lines are countable.

```suggestion
    This sniff verifies and auto-fixes the number of blank lines between property declarations within OO-structures.
```

   Previously this value would have to be set via a `config` directive in custom repo specific rulesets.
   For those rulesets which use the Yoast default, this `config` directive can now be removed.
   For more details, see [#131](https://github.com/Yoast/yoastcs/pull/131).
* PHPCS: All YoastCS native sniffs are now accompanied by documentation which can be viewed from the command-line using `phpcs --generator=Text --standard=Yoast`.
* Repo/QA: Various templates for typical pull requests to this repo.
* Composer: `fix-cs` script.
* Travis: Testing of the code against PHP 7.4 (unstable).

#### Changed
* PHPCS: Files in the following directories will now be excluded from all scans by default:
    - `/.git/`
    - `/.wordpress-svn/`
    - `/node-modules/`
    - `/vendor/`
    - `/vendor_prefixed/`
    Custom repo specific rulesets which contain excludes to this effect, can now remove them safely.
* PHPCS: The message type for issues reported by the `Generic.Formatting.MultipleStatementAlignment` and `WordPress.Arrays.MultipleStatementAlignment` sniffs, has been upgraded from `warning` to `error`.

> * PHPCS: The WPCS native check for the `json_encode()` function in the `WordPress.WP.AlternativeFunctions` has been disabled in favour of the new YoastCS native `Yoast.Yoast.AlternativeFunctions` sniff.

Reviewer: We should use American spelling.

```suggestion
* PHPCS: The WPCS native check for the `json_encode()` function in the `WordPress.WP.AlternativeFunctions` has been disabled in favor of the new YoastCS native `Yoast.Yoast.AlternativeFunctions` sniff.
```

* Composer: Supported version of [PHP_CodeSniffer] has been changed from `^3.4.0` to `^3.4.2`.
* Composer: Supported version of [WordPressCS] has been changed from `^2.0.0` to `^2.1.1`.
* Travis: As there is now a sniff which extends a WPCS sniff, the unit tests will now run against various combinations of PHPCS and WPCS combined.
* Minor housekeeping.

#### Removed
* PHPMD is no longer part of the YoastCS repo.
    PHPMD was not used as a stand-alone tool by any of the repos, only in combination with CodeClimate.
* Travis: Testing of the repo against PHP `nightly` (PHP 8, unstable) as no viable PHPUnit version is currently available.

#### Fixed

> * PHPCS: The `Yoast.Files.FileName` sniff will now always suggest to remove the longest prefix of the prefixes passed in the configuration.

Reviewer: Apparently "suggest to" is only for sentences in question form.

```suggestion
* PHPCS: The `Yoast.Files.FileName` sniff will now always suggest removing the longest prefix of the prefixes passed in the configuration.
```
